### PR TITLE
[BACKPORT] Backports Plugin Install Workflow and action for windows

### PR DESCRIPTION
**Pull request by scrawfor99**
### Description
Backports the plugin_install workflow and associated action. 

Backports  PR #2271 

Successfully ran CI and Plugin Install for Linux runner. Windows cannot be tested further until there is a Windows Min. Distribution available. 

### Check List
- [x] New functionality includes testing
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tools/install_demo_configuration.bat [hunk 1]
```diff
@@ -324,7 +324,7 @@ echo plugins.security.system_indices.indices: [".plugins-ml-model", ".plugins-ml
 	if %cluster_mode% == 1 (
         echo network.host: 0.0.0.0 >> "%OPENSEARCH_CONF_FILE%"
         echo node.name: smoketestnode >> "%OPENSEARCH_CONF_FILE%"
-        echo cluster.initial_master_nodes: smoketestnode >> "%OPENSEARCH_CONF_FILE%"
```
Comment (cwperks) [inline tools/install_demo_configuration.bat:327]: Can you remove the changes from this file?

Comment (scrawfor99) [reply tools/install_demo_configuration.bat:327]: Yeah I just brought it back but I can drop it.

**Comment by peternied** [comment]
> Windows cannot be tested further until there is a Windows Min. Distribution available.

@scrawfor99 Please reach out to the Engineering Effectiveness team to get help finding the min distro

File: .github/workflows/plugin_install.yml [hunk 2]
```diff
@@ -2,118 +2,60 @@ name: Plugin Install
...
-          'y' | .\Opensearch\bin\opensearch-plugin.bat install file:\a\security\opensearch-security.zip
-          'y', 'y', 'N' | .\Opensearch\plugins\opensearch-security\tools\install_demo_configuration.bat
+          echo at $(pwd)
+          ls
+          cd build
+          echo at $(pwd) 
+          ls
+          cd distributions
+          echo at $(pwd)
+          ls
```
Comment (peternied) [inline .github/workflows/plugin_install.yml:42]: Debug statements?

**Review by peternied — CHANGES_REQUESTED** [review]
Seems like this is still in draft while getting the windows min build, want to make that switch?  Note; I think it might be more useful to lump the changes for the plugin install workflow into a single PR that we approve at once rather than one at a time

File: .github/workflows/plugin_install.yml (hunk 2 above)
Comment (scrawfor99) [reply .github/workflows/plugin_install.yml:42]: Oops will remove.

**Comment by scrawfor99** [comment]
I rebased onto your branch and then PR'd against it. I am reaching out to the Engineering Effectiveness team now.

**Pull request closed without merging**